### PR TITLE
Synthetic internal powercell

### PR DESCRIPTION
**Pull request by QuickLode**
# About the pull request

This PR should tackle two things:
1. Synths shouldn't lose blood
2. Synths shouldn't feel woozy (RP)

revives #3589 from ghostsheet
# Explain why it's good for the game
Synthetics should not be feeling woozy. They also shouldn't be going blind from bloodloss. 
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

https://cdn.discordapp.com/attachments/490668342357786645/1157856898834497536/image.png?ex=651a21c4&is=6518d044&hm=813abcc8be5fd02367499e24651c97b7f840d8bcbc557364c534f1c0a61f25f5&

https://cdn.discordapp.com/attachments/490668342357786645/1157856899132305529/image.png?ex=651a21c4&is=6518d044&hm=d9fb14dcaa5eb9fa8ad1280636a5f8f188a72e0e5045b892fc29192046acd139&

</details>

# Changelog
:cl:
add: Damage to Synthetic's internal causes debuffs, eventual powercell failure.(death)
fix: Synthetic's should no longer lose blood.
code: moves blood proc to human.dm from blood.dm with above changes.
/:cl:

### CODE REVIEW

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/modules/mob/living/blood.dm [hunk 1]
```diff
@@ -47,27 +47,56 @@
 		switch(b_volume)
 			if(BLOOD_VOLUME_OKAY to BLOOD_VOLUME_SAFE)
 				if(prob(1))
-					var/word = pick("dizzy","woozy","faint")
-					to_chat(src, SPAN_DANGER("You feel [word]."))
+					low_blood_messsage(0)
```
Comment (BeagleGaming1) [inline code/modules/mob/living/blood.dm:50]: Do not use 0/1, use FALSE/TRUE

File: code/modules/mob/living/blood.dm [hunk 2]
```diff
@@ -47,27 +47,56 @@
...
-					var/word = pick("dizzy","woozy","faint")
-					to_chat(src, SPAN_DANGER("You feel extremely [word]."))
+					low_blood_messsage(2)
 			if(0 to BLOOD_VOLUME_SURVIVE)
-				death(create_cause_data("blood loss"))
+				death(create_cause_data(species.flags & IS_SYNTHETIC ? "power failure" : "blood loss"))
+
+
+		// Without enough blood you slowly go hungry.
+		if(blood_volume < BLOOD_VOLUME_SAFE)
```
Comment (morrowwolf) [inline code/modules/mob/living/blood.dm:73]: Shouldn't be re-added

File: code/modules/mob/living/blood.dm [hunk 3]
```diff
@@ -47,27 +47,56 @@
 		switch(b_volume)
 			if(BLOOD_VOLUME_OKAY to BLOOD_VOLUME_SAFE)
 				if(prob(1))
-					var/word = pick("dizzy","woozy","faint")
-					to_chat(src, SPAN_DANGER("You feel [word]."))
+					low_blood_messsage(0)
+				if(oxyloss < 20)
```
Comment (morrowwolf) [inline code/modules/mob/living/blood.dm:51]: Shouldn't be re-added

**Review by morrowwolf — CHANGES_REQUESTED** [review]

**Comment by Diegoflores31** [comment]
i though synths have milky blood . pretty sure winout it they would run non-optimally

File: code/modules/mob/living/blood.dm [hunk 4]
```diff
@@ -7,7 +7,7 @@
 
 // Takes care blood loss and regeneration
 /mob/living/carbon/human/handle_blood()
-	if(NO_BLOOD in species.flags)
+	if(species.flags & NO_BLOOD && !(species.flags & IS_SYNTHETIC))
```
Comment (realforest2001) [inline code/modules/mob/living/blood.dm:10]:
```suggestion
	if((species.flags & NO_BLOOD) && !(species.flags & IS_SYNTHETIC))
```

File: code/modules/mob/living/blood.dm [hunk 5]
```diff
@@ -91,7 +101,7 @@
 /mob/living/carbon/human/drip(amt)
 	if(in_stasis) // stasis now stops bloodloss
 		return
-	if(NO_BLOOD in species.flags)
+	if(species.flags & NO_BLOOD && !(species.flags & IS_SYNTHETIC))
```
Comment (realforest2001) [inline code/modules/mob/living/blood.dm:104]:
```suggestion
	if((species.flags & NO_BLOOD) && !(species.flags & IS_SYNTHETIC))
```

File: code/modules/mob/living/blood.dm [hunk 6]
```diff
@@ -282,6 +290,8 @@
 		return "whiteblood"
 	if(species.name == SPECIES_ZOMBIE)
 		return "greyblood"
+	if(species.flags & NO_BLOOD)
+		return
```
Comment (realforest2001) [inline code/modules/mob/living/blood.dm:294]: As this is an override it should remain at the top of the proc.

**Review by realforest2001 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/blood.dm [hunk 7]
```diff
@@ -50,6 +50,10 @@
 					var/word = pick("dizzy","woozy","faint")
 					to_chat(src, SPAN_DANGER("You feel [word]."))
 			if(BLOOD_VOLUME_BAD to BLOOD_VOLUME_OKAY)
+				if(species.flags & IS_SYNTHETIC)
+					if(prob(10))
+						apply_effect(rand(1,2), WEAKEN)
```
Comment (harryob) [inline code/modules/mob/living/blood.dm:55]:
```suggestion
						apply_effect(rand(1, 2), WEAKEN)
```

File: code/modules/mob/living/blood.dm [hunk 8]
```diff
@@ -58,16 +62,22 @@
...
-					AdjustEyeBlur(6)
-				oxyloss += 8
-				toxloss += 3
-				if(prob(15))
-					apply_effect(rand(1,3), PARALYZE)
-					var/word = pick("dizzy","woozy","faint")
-					to_chat(src, SPAN_DANGER("You feel extremely [word]."))
+				if(species.flags & IS_SYNTHETIC)
+					if(prob(10))
+						apply_effect(rand(1,2), PARALYZE)
```
Comment (harryob) [inline code/modules/mob/living/blood.dm:67]:
```suggestion
						apply_effect(rand(1, 2), PARALYZE)
```

File: code/modules/mob/living/blood.dm [hunk 9]
```diff
@@ -58,16 +62,22 @@
...
+						to_chat(src, SPAN_DANGER("Critical power cell failure detected.\nSeek recharging station immediately."))
+				else
+					if(eye_blurry < 50)
+						AdjustEyeBlur(6)
+						oxyloss += 8
+						toxloss += 3
+					if(prob(15))
+						apply_effect(rand(1,3), PARALYZE)
```
Comment (harryob) [inline code/modules/mob/living/blood.dm:75]:
```suggestion
						apply_effect(rand(1, 3), PARALYZE)
```

File: code/modules/mob/living/blood.dm [hunk 10]
```diff
@@ -58,16 +62,22 @@
...
+						var/word = pick("dizzy","woozy","faint")
```
Comment (harryob) [inline code/modules/mob/living/blood.dm:76]:
```suggestion
						var/word = pick("dizzy", "woozy", "faint")
```

**Review by harryob — CHANGES_REQUESTED** [review]
i can't help but feel this is making life code even more annoying but c'est la vie

File: code/modules/mob/living/blood.dm [hunk 11]
```diff
@@ -58,16 +62,22 @@
 					var/word = pick("dizzy","woozy","faint")
```
Comment (Diegoflores31) [inline code/modules/mob/living/blood.dm:62]: you forgot this part

File: code/modules/mob/living/blood.dm [hunk 12]
```diff
@@ -58,16 +62,22 @@
...
+						AdjustEyeBlur(6)
+						oxyloss += 8
+						toxloss += 3
+					if(prob(15))
+						apply_effect(rand(1, 3), PARALYZE)
+						var/word = pick("dizzy", "woozy", "faint")
+						to_chat(src, SPAN_DANGER("You feel extremely [word]."))
 			if(0 to BLOOD_VOLUME_SURVIVE)
-				death(create_cause_data("blood loss"))
+				death(create_cause_data(species.flags & IS_SYNTHETIC ? "power failure" : "blood loss"))
```
Comment (Diegoflores31) [inline code/modules/mob/living/blood.dm:79]: power failure its a little confusing i would recomend something more descriptive ?

File: code/modules/mob/living/blood.dm [hunk 13]
```diff
@@ -58,16 +62,22 @@
...
-				oxyloss += 8
-				toxloss += 3
-				if(prob(15))
-					apply_effect(rand(1,3), PARALYZE)
-					var/word = pick("dizzy","woozy","faint")
-					to_chat(src, SPAN_DANGER("You feel extremely [word]."))
+				if(species.flags & IS_SYNTHETIC)
+					if(prob(10))
+						apply_effect(rand(1, 2), PARALYZE)
+						to_chat(src, SPAN_DANGER("Critical power cell failure detected.\nSeek recharging station immediately."))
```
Comment (Diegoflores31) [inline code/modules/mob/living/blood.dm:68]: sparks !!

**Review by Diegoflores31 — COMMENTED** [review]
along with that since you are touching their blood .  when a synth looses all its blood same than a human it will instantly die . i would recomend adding maybe a "synthetic blood" to medbay or something so it can be replenished ?

**Review by fira — APPROVED** [review]

File: code/modules/mob/living/blood.dm (hunk 11 above)
Comment (harryob) [reply code/modules/mob/living/blood.dm:62]: it'll still say "You feel very dizzy/woozy/faint"

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by harryob** [comment]
really the whole handle_blood proc should be offlaoded to /datum/species for /humans

File: code/modules/mob/living/blood.dm [hunk 14]
```diff
@@ -50,24 +50,35 @@
 					var/word = pick("dizzy","woozy","faint")
```
Comment (Diegoflores31) [inline code/modules/mob/living/blood.dm:50]: you forgor here too

**Review by Zonespace27 — COMMENTED** [review]
Balance fine; the code for this is a bit gross but 🤷

**Pull request merged**